returnn/torch/engine.py

```diff
@@ -380,6 +384,14 @@ def train_epoch(self):
                     torch.distributed.all_reduce(_has_data, op=torch.distributed.ReduceOp.MIN)
                 if not _has_data[0]:
                     break
+
+                keys_w_seq_len = [k for k in extern_data_raw if f"{k}:seq_len" in extern_data_raw]
+                data_len = sum(torch.sum(extern_data_raw[f"{k}:seq_len"]) for k in keys_w_seq_len)
+                data_space = sum(torch.numel(extern_data_raw[k]) for k in keys_w_seq_len)
```

Comment:
Despite the `sum` (over the keys), which you should not do as discussed above, `numel` is also wrong here? That would also count the feature dim(s)? You want to compare sum(seq_len[k]) vs prod(data[k].shape[:2]).